src/environment.cpp

```diff
@@ -1256,6 +1256,50 @@ void ServerEnvironment::step(float dtime)
 		*/
 		removeRemovedObjects();
 	}
+
+	/*
+		Manage particle spawner expiration
+	*/
+	if (m_particle_management_interval.step(dtime, 1.0)) {
+		for(std::map<u32, float>::iterator i =
+		    m_particle_spawners.begin();
+		    i != m_particle_spawners.end();) {
+			//non expiring spawners
+			if (i->second <= -1.f) {
+				i++;
+				continue;
+			}
+
+			i->second -= 1.0;
+			if (i->second < 0.f)
+				m_particle_spawners.erase(i++);
+			else
+				i++;
```

Comment:
so after reading up on this i wonder if this is correct, ++i is suggested in order not to skip any elements. also i think zenos 2nd version above is clearer than mine but has the same postfix increment, so im commenting here